Client/Frontend/Browser/HomePanel/NTPDownloader.swift

```diff
@@ -0,0 +1,378 @@
+}
+
+struct NTPWallpaper: Codable {
+    let imageUrl: String
+    let focalPoint: NTPFocalPoint?
+}
+
+struct NTPItemInfo: Codable {
+    let logo: NTPLogo
+    let wallpapers: [NTPWallpaper]
+}
+
+class NTPDownloader {
+    private static let etagFile = "crc.etag"
+    private static let metadataFile = "photo.json"
+    private static let ntpDownloadsFolder = "NTPDownloads"
+    private static let baseURL = "https://brave-ntp-crx-input-dev.s3-us-west-2.amazonaws.com/"
+    
+    private let isZipped: Bool
+    private let defaultLocale = "US"
+    private let supportedLocales: [String] = []
+    private let currentLocale = Locale.current.regionCode
+    
+    init(isZipped: Bool) {
+        self.isZipped = isZipped
+    }
+    
+    func getNTPInfo(_ completion: @escaping (NTPItemInfo?) -> Void) {
+        // Download the NTP Info to a temporary directory
+        self.download { [weak self] url, cacheInfo, error in
+            guard let self = self else { return }
+            
+            if let error = error {
+                logger.error(error)
+                return completion(self.loadNTPInfo())
+            }
+            
+            guard let url = url else {
+                logger.error(error)
+                return completion(self.loadNTPInfo())
+            }
+            
+            if let cacheInfo = cacheInfo, cacheInfo.statusCode == 304 {
+                logger.debug("NTPDownloader Cache is still valid")
+                return completion(self.loadNTPInfo())
+            }
+            
+            //Move contents of `url` directory
+            //to somewhere more permanent where we'll load the images from..
+            guard let supportDirectory = FileManager.default.urls(for: .applicationSupportDirectory, in: .userDomainMask).first else {
+                return completion(self.loadNTPInfo())
+            }
+ 
+            do {
+                let downloadsURL = supportDirectory.appendingPathComponent(NTPDownloader.ntpDownloadsFolder)
+                if FileManager.default.fileExists(atPath: downloadsURL.path) {
+                    try FileManager.default.removeItem(at: downloadsURL)
+                }
+                
+                try FileManager.default.moveItem(at: url, to: downloadsURL)
+                
+                //Store the ETag
+                if let cacheInfo = cacheInfo {
+                    self.setETag(cacheInfo.etag)
+                }
+                
+                completion(self.loadNTPInfo())
+            } catch {
+                logger.error(error)
+                completion(self.loadNTPInfo())
+            }
+        }
+    }
+    
+    private func loadNTPInfo() -> NTPItemInfo? {
+        guard let supportDirectory = FileManager.default.urls(for: .applicationSupportDirectory, in: .userDomainMask).first else {
+            return nil
+        }
+        
+        let downloadsURL = supportDirectory.appendingPathComponent(NTPDownloader.ntpDownloadsFolder)
```

Comment:
nitpick: I feel it's easier to read if you name it `downloadsFolder` not `URL`, same for metaDataUrl etagURL

Reply:
Fixed.